cpp/tests/utilities/type_lists.hpp

```diff
@@ -101,11 +113,23 @@ using TimestampTypes = cudf::test::Types<timestamp_D, timestamp_s, timestamp_ms,
  *---------------------------------------------------------------------------**/
 using FixedWidthTypes = Concat<NumericTypes, TimestampTypes>;
 
+/**---------------------------------------------------------------------------*
+ * @brief Provides a list of all fixed-width element types for use in GTest
+ * typed tests.
+ * 
+ * Example:
+ * ```
+ * // Invokes all typed fixture tests for all fixed-width types in libcudf
+ * TYPED_TEST_CASE(MyTypedFixture, cudf::test::FixedWidthTypes);
+ * ```
+ *---------------------------------------------------------------------------**/
+using LexicographicalTypes = Concat<NumericTypes, TimestampTypes, StringTypes>;
```

Comment:
Instead of adding a new type list, why not add `StringTypes` to `AllTypes`?

Reply:
Practically, it breaks other tests, and I feel fixing those tests falls outside the scope of this PR. Conceptually, it's conceivable not all types will be lexicographical, and therefore having a type list specific to lexicographical types prevents us from having to changes these tests in the future if a non-sortable type is added to `AllTypes` (which is similar to the problem I'm seeing when adding `StringTypes` to `AllTypes`.

TLDR; `StringTypes` should probably be added to `AllTypes`, but it would require making sure some unrelated tests are updated.

Reply:
Makes sense. I'd just suggest renaming `LexicographicalTypes` to `ComparableTypes`.

Reply:
Done.